### DISCUSSION TITLE
Release 1.7.1/1.8.0

### DISCUSSION THREAD

**Discussion by whalley** [moneymanagerex/moneymanagerex#6518] [Development]
There were a few crash reports soon after the v1.7.1 release that have resulted in a few fixes. I think it would therefore be wise to not wait too long before we issue v1.7.1. I'm thinking that we push out a release in a few weeks, certainly before mid-March.

Views?

**Comment by renato-mmex** [comment]
I would welcome it if we could release 1.7.1 in mid-March.
Wouldn't it make sense to check the still open problems for urgency and postpone less important ones to 1.7.2?

**Reply by whalley** [reply]
Yep, that is the next step if we agree to release next month.

**Comment by n-stein** [comment]
Sounds good. I'll focus my efforts on bug fixes rather than features or enhancements.

**Comment by whalley** [comment]
Think we need to wait a few more weeks to test recent changes... Have bumped version to RC.1

**Reply by renato-mmex** [reply]
Which changes are meant by this?

**Reply by whalley** [reply]
Timed transactions in particular, we merged some changes by @vomikan very recently.

**Comment by vomikan** [comment]
Don't you think this version should be 1.8.0?

**Reply by whalley** [reply]
Possibly... I guess the feature of timed transactions may indeed warrant this.

**Comment by renato-mmex** [comment]
is there already a date for version 1.7.1

**Reply by whalley** [reply]
I've been away for the past week or so....  @n-stein @vomikan Is the release now stable?

**Reply by n-stein** [reply]
Seems fairly stable, but I would like to investigate this reported bug before release:

https://forum.moneymanagerex.org/viewtopic.php?t=11065

**Comment by renato-mmex** [comment]
From my point of view, you could move https://github.com/moneymanagerex/moneymanagerex/issues/6503 to 1.7.2 and release the version 1.7.1. 
@whalley @n-stein @vomikan

**Reply by n-stein** [reply]
#6503 is complete, although I did just push a minor revision to only reset the panel column order after the options dialog is closed rather than every time you change accounts (#6655). Otherwise, your column order doesn't persist when switching between accounts of the same type like it did in 1.7.0.

The only thing holding up 1.7.1 is which, if any, of the recently logged bugs are significant enough to delay the release. We had about 10 or so pop up in the last week, and I've only corrected a few that required very minor changes. The only one I think should be addressed before release is #6637 (which is the same bug I mentioned as posted in the forum https://forum.moneymanagerex.org/viewtopic.php?t=11065), but I haven't been able to reproduce it.

**Reply by whalley** [reply]
Been away, have merged the recent changes. Suggest we aim for release at end of this month.

**Comment by renato-mmex** [comment]
Since the user has not responded to the error message #6637  and the error is not reproducible, I see no reason to wait with the release of 1.7.1.

**Comment by whalley** [comment]
We good for new release this weekend? Should we go with v1.8.0 as @vomikan suggested, given the new 'time-based' transaction reporting being a major new feature

**Reply by ovari** [reply]
Does SemVer mean the version should be MAJOR.MINOR.PATCH?

> given the new 'time-based' transaction reporting being a **major** new feature

Should the version change from v1.7.0 to v2.0.0?

To make up for major features, should the version be changed from v1.7.0 to v19.0.0?

Thank you

**Comment by renato-mmex** [comment]
This would mean that all reports from 1.7.2 (80) + 1.7.3 (87) would be postponed to 1.8. These 167 open cases + the new urgent cases would then have to be resolved relatively quickly, otherwise 1.8 will not be completed this year.

**Reply by whalley** [reply]
They will just move to 1.8.1 and 1.8.2

**Comment by vomikan** [comment]
I've renamed milestones 1.7.1 to 1.8.0 and 1.7.2 to 1.8.1 ....

**Comment by ovari** [comment]
@joshua-stone FYI [v1.8.0](https://github.com/moneymanagerex/moneymanagerex/milestone/42) due by April 25, 2024

**Reply by joshua-stone** [reply]
I'm going to try building it now:

https://github.com/flathub/org.moneymanagerex.MMEX/pull/17

**Reply by ovari** [reply]
@joshua-stone is a rebuild required due to https://github.com/moneymanagerex/moneymanagerex/pull/6679?

**Reply by whalley** [reply]
The CI should build this anyway... 
https://app.circleci.com/pipelines/github/moneymanagerex/moneymanagerex/1412/workflows/f93d2784-a99c-45d8-b2bc-4c7f7b6cf72c/jobs/1405/artifacts

**Reply by joshua-stone** [reply]
Looks like the latest commit built successfully:

https://buildbot.flathub.org/#/builders/6/builds/116972

I think we should be ready to tag a new version.

**Comment by whalley** [comment]
v1.8.0 is released, @joshua-stone could you please ensure Flathub is updated.

**Reply by ovari** [reply]
@whalley can the [pull requests](https://github.com/moneymanagerex/moneymanagerex/pulls) with `on hold` tags please be merged? If not all at the moment, can the ones with the `v1.8.1` tags please be merged? Thank you
